vetomint/tests/integration_test.rs

```diff
@@ -83,3 +85,594 @@ fn success_trivial_1() {
         vec![ConsensusResponse::FinalizeBlock { proposal: 0 }]
     );
 }
+
+#[ignore]
+#[test]
+//New block with 2/3+ e.t., ununiformed voting-power and pre-committed
+fn new_block_et_two_thirds_precommitted2() {
+    let height_info = HeightInfo {
+        validators: vec![10, 8, 6, 5, 4, 2, 2],
+        this_node_index: 3,
+        timestamp: 0,
+        consensus_params: ConsensusParams { timeout_ms: 1000 },
+    };
+    let mut state = ConsensusState::new(height_info.clone());
+
+    // STEP 1: Proposal.
+    let event = ConsensusEvent::BlockProposal {
+        proposal: 0,
+        proposer: 0,
+        round: 0,
+        time: 1,
+    };
+    let response = progress(&height_info, &mut state, event);
+    assert!(response.is_empty());
+    let event = ConsensusEvent::ProposalFavor {
+        proposal: 0,
+        favor: true,
+        time: 2,
+    };
+    let response = progress(&height_info, &mut state, event);
+    assert_eq!(
+        response,
+        vec![ConsensusResponse::BroadcastPrevote {
+            proposal: 0,
+            round: 0
+        }]
+    );
+
+    // STEP 2: Prevote.
+    for validator_index in 0..=1 {
+        let event = ConsensusEvent::Prevote {
+            proposal: 0,
+            round: 0,
+            signer: validator_index,
+            time: 3,
+        };
+        let response = progress(&height_info, &mut state, event);
+        assert!(response.is_empty());
+    }
+    let event = ConsensusEvent::Prevote {
```

Comment:
If nodes 0, 1, and 2 prevote, 
the total voting power is 24 / 37 < 2/3 so it will not terminate

Reply:
> If nodes 0, 1, and 2 prevote, the total voting power is 24 / 37 < 2/3 so it will not terminate

Since this_node_index is 3 and it favors this proposal, total voting power for this proposal is 10+8+6+5=29 > 2/3